### PR TITLE
DO_NOT_MERGE Exercise 6

### PR DESCRIPTION
**Pull request by ashu26jha**
New Sync package using feeder gateway

### CODE REVIEW

File: sync/sync.go [hunk 1]
```diff
@@ -297,6 +307,138 @@ func (s *Synchronizer) syncBlocks(syncCtx context.Context) {
...
+	streamCtx, streamCancel := context.WithCancel(syncCtx)
+	for {
+		select {
+		case <-streamCtx.Done():
+			streamCancel()
+			return
+		default:
+			s.log.Infow("Fetching from feeder")
+			blocks, stateUpdate, blockCommitments := s.getBlockNumberDetails(uint64(s.latestBlockHeight))
+			newClass, _ := s.fetchUnknownClasses(syncCtx, &stateUpdate)
```
Comment (rianhughes) [inline sync/sync.go:320]: You should handle the error

**Comment by rianhughes** [comment]
Generally it looks pretty good, although I would recommend fixing the tests.

File: sync/sync.go (hunk 1 above)
Comment (ashu26jha) [reply sync/sync.go:320]: I will try to fix this through the manually, if it doesn't work I will switch to the starknetdata interface as you mentioned above!

**Comment by ashu26jha** [comment]
Hey @rianhughes, did the changes  I am able to sync but there are couple of flaws I would like to point out:

1. Can't fix all the lints as http is not added to ignore 
2. It seems to run indefinitely
3. It only works with mainnet

Should I try fixing these or move on?

**Comment by kirugan** [comment]
Closing since it was onboarding exercise

**Pull request closed without merging**